### ISSUE TITLE
[ICE] with const generics in subst.rs

### ISSUE THREAD

**Issue by leonardo-m** [rust-lang/rust#71922]
This is wrong code because it lacks the const_generics feature request, and because currently calling a const function in this context isn't allowed:

```rust
const fn foo(n: usize) -> usize { n * 2 }
fn bar<const N: usize>() -> [u32; foo(N)] {
    [0; foo(N)]
}
fn main() {}
```

Gives:

```
error[E0658]: const generics are unstable
 --> ...\test.rs:2:14
  |
2 | fn bar<const N: usize>() -> [u32; foo(N)] {
  |              ^
  |
  = note: see issue #44580 <https://github.com/rust-lang/rust/issues/44580> for more information
  = help: add `#![feature(const_generics)]` to the crate attributes to enable

error: internal compiler error: src\librustc_middle\ty\subst.rs:581: const parameter `N/#0` (Const { ty: usize, val: Param(N/#0) }/0) out of range when substituting substs=[]

thread 'rustc' panicked at 'Box<Any>', /rustc/2454a68cfbb63aa7b8e09fe05114d5f98b2f9740\src\libstd\macros.rs:13:23
stack backtrace:
  ...

note: rustc 1.45.0-nightly (2454a68cf 2020-05-04) running on x86_64-pc-windows-gnu

query stack during panic:
#0 [const_eval_raw] const-evaluating `bar::{{constant}}#0`
#1 [const_eval_validated] const-evaluating + checking `bar::{{constant}}#0`
#2 [const_eval_validated] const-evaluating + checking `bar::{{constant}}#0`
#3 [check_item_well_formed] processing `bar`
#4 [analysis] running analysis passes on this crate
end of query stack
error: aborting due to 2 previous errors
```

**Comment by lcnr** [comment]
blocked on lazy normalization #67890.

Probably a duplicate of https://github.com/rust-lang/rust/issues/69913

**Comment by lcnr** [comment]
This issue still isn't fixed :thinking: It seems like I made a mistake in #71973

**Comment by JohnTitor** [comment]
Triage: This is no longer ICE (even with `const_generics`) on the latest nightly: https://play.rust-lang.org/?version=nightly&mode=debug&edition=2018&gist=f0afc6e029354b7b4e933cb00c9b0cba

**Comment by lcnr** [comment]
we already have a bunch of similar tests, closing without adding an additional one
